arbor/profile/profiler.cpp

```diff
@@ -96,9 +100,14 @@ class profiler {
     // Used to protect name_index_, which is shared between all threads.
     std::mutex mutex_;
 
+    // Flag to indicate whether the profiler has been initialized with the task_system
+    bool init_ = false;
+
 public:
     profiler();
+    profiler(unsigned nthreads);
```

Comment:
I think the profiler should simply default construct to uninitialized state. A constructor that takes a number of threads while leaving it uninitialized is a bit confusing.